src/miner.cpp

```diff
@@ -164,13 +164,13 @@ CBlockTemplate* CreateNewBlock(const CScript& scriptPubKeyIn)
     pblocktemplate->vTxSigOps.push_back(-1); // updated at end
 
     // Largest block you're willing to create:
-    unsigned int nBlockMaxSize = GetArg("-blockmaxsize", MAX_BLOCK_SIZE_GEN/2);
+    unsigned int nBlockMaxSize = GetArg("-blockmaxsize", MAX_BLOCK_SIZE);
```

Comment:
Maybe we should create GetMandatoryArg versions that throw exceptions or assert-fail when required values are not provided ? If you do this, please the global argMap explictly as a const ref argument in the new versions, like in #6423